server/src/test-fast/java/com/thoughtworks/go/config/ConfigConverterTest.java

```diff
@@ -162,6 +164,24 @@ public void shouldConvertEnvironmentVariableWhenSecure() {
         assertThat(result.getName(), is("key1"));
     }
 
+    @Test
+    public void shouldConvertEnvironmentVariableToSecureWhenEncryptedValueIsEmptyString() {
+        CREnvironmentVariable crEnvironmentVariable = new CREnvironmentVariable("key1", null, "");
+        EnvironmentVariableConfig result = configConverter.toEnvironmentVariableConfig(crEnvironmentVariable);
+        assertThat(result.isSecure(), is(true));
+        assertThat(result.getValue(), is(""));
+        assertThat(result.getName(), is("key1"));
+    }
+
+    @Test
+    public void shouldConvertEnvironmentVariableToInsecureWhenEncryptedValueIsNull() {
```

Comment:
Just a thought that def doesn't need to be done, but maybe this test can be merged with `#shouldConvertNullEnvironmentVariableWhenNotSecure` ?

They're basically doing the same thing, and it'd be one less test?